app/javascript/react/containers/NewAirportFormContainer.js

```diff
@@ -0,0 +1,169 @@
+    this.addNewAirport = this.addNewAirport.bind(this)
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.name]: event.target.value })
+  }
+
+  addNewAirport(formPayload){
+    fetch("/api/v1/airports",{
+      credentials: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify(formPayload),
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+        }
+      })
+      .then(response => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status}(${response.statusText})` ,
+          error = new Error(errorMessage);
+          throw(error);
+        }
+        })
+        .then(response => {
+          browserHistory.push('/airports')
+        })
+        .catch(error => console.error(`Error in fetch: ${error.message}`));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault()
+
+    if (
+      this.validateNameInput(this.state.airportName) &&
+      this.validateLocationInput(this.state.airportLocation) &&
+      this.validateAirportCodeInput(this.state.airportCode)
+    ) {
+      let formPayload = {
+        name: this.state.airportName,
+        location: this.state.airportLocation,
+        airport_code: this.state.airportCode,
+        description: this.state.airportDescription
+      }
+      this.addNewAirport(formPayload);
+      this.handleClearForm(event)
+    }
+  }
+
+  handleClearForm(event) {
+    event.preventDefault()
+    this.setState({
+      airportName: '',
+      airportLocation: '',
+      airportCode: '',
+      airportDescription: '',
+      errors: {}
+    })
+  }
+
+  validateNameInput(selection) {
+    if (selection.trim() === '') {
+      let newError = { nameError: 'You must enter a name for the airport.' }
+      this.setState({ errors: Object.assign({}, this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.nameError
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateAirportCodeInput(selection) {
+    if (selection.trim() === '') {
+      let newError = { codeError: 'You must enter a 3 letter code for the airport.' }
+      this.setState({ errors: Object.assign({}, this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.codeError
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateLocationInput(selection) {
+    if (selection.trim() === '') {
+      let newError = { locationError: 'You must enter a location for the airport.' }
+      this.setState({ errors: Object.assign({}, this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.locationError
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
```

Comment:
Your validate methods look to be nearly identical. I think you should find a way to make one method to handle validation and change how you're passing the selection into it to identify which thing you're validating.

Reply:
Condensed to a single function

Reply:
Great job!